articles/quickstart/spa/angular-beta/_includes/_centralized_login.md

```diff
@@ -0,0 +1,448 @@
+
+  /**
+   * Constructor - inject the AuthService class
+   */
+  constructor(private authService: AuthService) {}
+
+  /**
+   * Handle component initialization
+   */
+  async ngOnInit() {
+    // Get an instance of the Auth0 client
+    this.auth0Client = await this.authService.getAuth0Client();
+
+    // Watch for changes to the isAuthenticated state
+    this.authService.isAuthenticated.subscribe(value => {
+      this.isAuthenticated = value;
+    });
+
+    // Watch for changes to the profile data
+    this.authService.profile.subscribe(profile => {
+      this.profile = profile;
+    });
+  }
+
+  /**
+   * Logs in the user by redirecting to Auth0 for authentication
+   */
+  async login() {
+    await this.auth0Client.loginWithRedirect({
+      redirect_uri: `<%= "${window.location.origin}" %>/callback`
+    });
+  }
+
+  /**
+   * Logs the user out of the applicaion, as well as on Auth0
+   */
+  logout() {
+    this.auth0Client.logout({
+      client_id: this.authService.config.client_id,
+      returnTo: window.location.origin
+    });
+  }
+}
+```
+
+Notice that the `AuthService` class you created in the previous section is being injected into the component through the constructor. This allows you to get an instance of the client, as well as react to changes in authentication state.
+
+The main setup work is being carried out inside `ngOnInit`, where the RxJS `BehaviorSubject` instances are being used. Whenever the values of these change, the Navbar component will react to those changes and provide an updated UI.
+
+Functions are provided to log in and log out the user using `loginWithRedirect` and `logout` directly on the Auth0 client. In the login scenario, a URI is provided that indicates where Auth0 should redirect to once authentication is complete. For login to work, this URL must be specified as an **Allowed Callback URL** in your [application settings](${manage_url}/#/applications/${account.clientId}/settings).
+
+Next, configure the UI for the `navbar` component by opening the `src/app/navbar/navbar.component.html` file and replacing its contents with the following:
+
+```html
+<header>
+  <button (click)="login()" *ngIf="!isAuthenticated">Log In</button>
+  <button (click)="logout()" *ngIf="isAuthenticated">Log Out</button>
+</header>
+```
+
+Angular bindings are used here to show each button at the right time; the Login button when the user is unauthenticated, and the Log Out button when the user has been authenticated. Their `click` events are wired up to call the `login` and `logout` functions on the component respectively.
+
+Finally, to show this component on the screen, open the `src/app/app.component.html` file and replace the default UI with the following:
+
+```html
+<app-navbar></app-navbar>
+
+<router-outlet></router-outlet>
+```
+
+> **Checkpoint**: Run the application now using `npm start`. The login button should be visible, and you should be able to click it to be redirected to Auth0 for login.
```

Comment:
@joshcanhelp I've added a section in above that shows how to configure the app so that it runs on the correct port (3000), with a note minding them to correct the URLs if they're using their own application with a different port.